src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -640,6 +647,52 @@ private static void SetChunkedModeForSend(HttpRequestMessage request)
             }
         }
 
+        private static Guid LogRequest(HttpRequestMessage request)
+        {
+            if (_diagnosticSource.IsEnabled("System.Net.Http.Request"))
+            {
+                Guid loggingRequestId = Guid.NewGuid();
+                long timestamp = Stopwatch.GetTimestamp();
+
+                _diagnosticSource.Write(
+                    "System.Net.Http.Request",
+                    new
+                    {
+                        Request = request,
+                        LoggingRequestId = loggingRequestId,
+                        Timestamp = timestamp
+                    });
+
+                return loggingRequestId;
+            }
+
+            return Guid.Empty;
+        }
+
+        private static void LogResponse(Task<HttpResponseMessage> response, Guid loggingRequestId)
+        {
+            if (_diagnosticSource.IsEnabled("System.Net.Http.Response"))
+            {
+                if (!response.IsFaulted &&
+                    !response.IsCanceled &&
+                    // An empty loggingRequestId signifies that the request was not logged, so do
+                    // not attempt to log response.
+                    loggingRequestId != Guid.Empty)
+                {
+                    long timestamp = Stopwatch.GetTimestamp();
+
+                    _diagnosticSource.Write(
+                        "System.Net.Http.Response",
+                        new
+                        {
+                            Response = response.Result,
+                            LoggingRequestId = loggingRequestId,
+                            Timestamp = timestamp
+                        });
+                }
+            }
+        }
+
```

Comment:
Fixed (shared code for this moved to :/src/common/src/system/net/http/httphandlerdiagnosticlistenerextension.cs).